### DISCUSSION TITLE
Training with own data

### DISCUSSION THREAD

**Discussion by Shin-ichi-Takayama** [speechbrain/speechbrain#1622] [Q&A] [answered]
Thank you for all your help.
I would like to train with my own Japanese data in Sepformer.
Do you have any documentation explaining the procedure and the composition of the training data?

**Comment by Adel-Moumen** [comment]
Can you take a look, please @ycemsubakan, thanks!  :-)

**Comment by ycemsubakan** [comment, marked as answer]
Hi @Shin-ichi-Takayama , 

What you need to change compared to the WSJ0Mix recipe is simply the `.csv` files that points to the data paths. You can use the .yaml file below for this: 

https://github.com/speechbrain/speechbrain/blob/develop/recipes/WSJ0Mix/separation/hparams/sepformer-customdataset.yaml

Basically, if you do not include the letters `wsj` in your `--datafolder`  (see https://github.com/speechbrain/speechbrain/blob/64c8fa951bdade7e28c81b37452331c61af7b548/recipes/WSJ0Mix/prepare_data.py#L36), the code will use this function below to create the `.csv` file for data paths.

https://github.com/speechbrain/speechbrain/blob/64c8fa951bdade7e28c81b37452331c61af7b548/recipes/WSJ0Mix/prepare_data.py#L55)

You can simply create your data paths in accordance with this function. 

Let me know if you have any questions.

**Reply by Shin-ichi-Takayama** [reply]
@ycemsubakan 
Thank you for your response.
I now understand how to study with my own data.
If I have any questions, I will ask again.

**Reply by Shin-ichi-Takayama** [reply]
@ycemsubakan 
Thank you for your assistance.
Thanks to your advice I was able to create a custom data set.
I ran the training with the following commands

> python train.py hparams/sepformer-customdataset.yaml --data_folder ../data/

However, I got the following error message.

> speechbrain.core - Beginning experiment!
speechbrain.core - Experiment folder: results/sepformer-custom/1234
speechbrain.core - Exception:
Traceback (most recent call last):
  File "train.py", line 541, in <module>
    from recipes.WSJ0Mix.prepare_data import prepare_wsjmix  # noqa
ModuleNotFoundError: No module named 'recipes'

Is the training method incorrect?

**Comment by ycemsubakan** [comment]
It seems you haven't installed speechbrain. Can you install speechbrain and try again?

**Reply by Shin-ichi-Takayama** [reply]
@ycemsubakan 
Thank you for your response.
I checked to see if Speechbrain was installed using the `pip list` command.
It was installed as follows

> pip list | grep speechbrain
speechbrain         0.5.13          /root/speechbrain

I installed it with github using the following command

> git clone https://github.com/speechbrain/speechbrain.git
cd speechbrain
pip install -r requirements.txt
pip install --editable .

Do I need to install via PyPI?
The install test passed.

**Reply by ycemsubakan** [reply]
Hi @Shin-ichi-Takayama ,

If you now go to `recipes/WSJ0Mix/separation` under speechbrain repo that you have cloned, when you run `train.py`, you should be able to import `prepare_wsjmix` function without any errors.
Is it the case?

**Reply by Shin-ichi-Takayama** [reply]
@ycemsubakan 
Thank you for your reply.
I added the cloned repository to the `PYTHONPATH` as follows and it worked.

> export PYTHONPATH="/root/speechbrain:$PYTHONPATH"

I found what seems to be a bug in the `create_custom_dataset` function in `prepare_data.py` when I trained with my own data.
I will report the bug later.
